### PR TITLE
A simple event emitter for Game

### PR DESCRIPTION
**Pull request by stasm**

### CODE REVIEW

File: core/game.js [hunk 1]
```diff
@@ -135,42 +131,48 @@ export class Game {
...
-      if (this.actions[action]) {
-        this.actions[action](tick_time);
-      }
-    });
-
-    this.entities.forEach((entity) => entity.update(this.tick_length));
-
-    this.camera.update(this.tick_length);
+  update() {
+    this.emit('tick', this.last_tick);
```
Comment (stasm) [inline core/game.js]: I wonder if we should perhaps rename `update` to `tick`, or the `tick` event to `update`?  That's a huge _I wonder_; I'm happy to leave this as-is.

File: core/game.js [hunk 2]
```diff
@@ -85,32 +84,29 @@ export class Game {
...
     }
+  }
 
-    const next_tick = this.last_tick + this.tick_length;
-
-    if (tick_time > next_tick) {
-      const ticks_qty = Math.floor((tick_time - this.last_tick) / this.tick_length);
-      this.perform_ticks(ticks_qty);
-      this.render(this.last_tick);
+  add_listener(event_name, handler) {
```
Comment (michalbe) [inline core/game.js]: What do you think about names like `on` and `off`? It's a common pattern among js libraries and should save us few bytes :).

Comment (stasm) [reply core/game.js]: Sure, I like it!  The shorter the names of the public API, the better!

File: tweens/basic-tween.js [hunk 3]
```diff
@@ -34,7 +34,7 @@ export class BasicTween {
       this.do_step.bound = (tick) => {
         this.do_step(tick, resolve);
       };
-      this.game.add_frame_action(this.do_step.bound);
+      this.game.on("tick", this.do_step.bound);
```
Comment (michalbe) [inline tweens/basic-tween.js]: NIT: single quotes pls.

**Pull request merged**